clis/teliod/src/main.rs

```diff
@@ -96,13 +105,21 @@ async fn main() -> Result<(), TeliodError> {
             if DaemonSocket::get_ipc_socket_path()?.exists() {
                 Err(TeliodError::DaemonIsRunning)
             } else {
+                #[cfg(feature = "qnap")]
+                let _pid_guard = qnap::PidFile::new().await?;
                 let file = File::open(&config_path)?;
+
                 let mut config: TeliodDaemonConfig = serde_json::from_reader(file)?;
-                let token = std::env::var("NORD_TOKEN").ok();
-                if let Some(t) = token {
+
+                if let Ok(token) = std::env::var("NORD_TOKEN") {
                     debug!("Overriding token from env");
-                    config.authentication_token = t;
+                    if token.len() == 64 && token.chars().all(|c| c.is_ascii_hexdigit()) {
```

Comment:
It would be nice to make AuthorizationToken type which would enforce these environments. e.g. something on the lines like:

```rust
struct AuthorizationToken {
...
};

impl AuthorizationToken {
    pub fn new(String token) -> Self {...}
}
```

And then every place, which needs access to AuthorizationToken would use `AuthorizationToken` struct instead of String. The benefit of such approach is that we can be sure (in a type safe manner) that invariants (e.g. length and symbol requirements) are always enforced for such type, and thus it is significantly harder to miss validation.

Reply:
I had pretty much the same thought here: https://github.com/NordSecurity/libtelio/pull/1012/files#r1894030576

Reply:
Agreed! However this will be part of the next PR.